docs/app-dev-spark-connect.md

```diff
@@ -58,6 +58,59 @@ to the client via the blue box as part of the Spark Connect API. The client uses
 alongside PySpark or the Spark Scala client, making it easy for Spark client applications to work
 with the custom logic/library. 
 
+## Spark API Mode: Spark Client and Spark Classic
+
+Spark provides API mode (`spark.api.mode`) configuration that can seamlessly start Spark Classic
+applications to use Spark Connect. Based on `spark.api.mode` configuration, the application either
+be executed in Spark Classic or Spark Connect mode. See the examples below:
+
+{% highlight python %}
+from pyspark.sql import SparkSession
+
+SparkSession.builder.config("spark.api.mode", "connect").master("...").getOrCreate()
+{% endhighlight %}
+
+
+This configuration can be also effectively set for both Scala Spark application and PySpark
+applications via Spark submission:
+
+{% highlight bash %}
+spark-submit --master "..." --conf spark.api.mode=connect
+{% endhighlight %}
+
+In addition, Spark Connect provides convient special cases for you to locally test. `spark.remote`
+can be set as `local[...]` and `local-cluster[...]`. In this case, it creates a locally running
+Spark Connect server, and provide users the Spark Connect session. This is similar when
+`--conf spark.api.mode=connect` and `--master ...` are set together. However `spark.remote`
+and `--remote` only supports `local*` whereas `--conf spark.api.mode=connect` and `--master ...`
+allow other Spark cluster URLs such as `spark://` for better compatibility.
```

Comment:
@hvanhovell this part is slightly different from what we discussed offline. My logic here is to restrict it first instead of allowing it: `--remote` and `spark.remote` will support only `local*` as it was.